docs/tools/near-indexer.md

```diff
@@ -0,0 +1,487 @@
+
+### Setup
+
+The node will need three things:
+
+1) Configs
+2) Network's genesis file
+3) Key to run the node
+
+All of this can be generated via nearcore's `neard` crate, but an easier way is to use the exposed `init_configs` from NEAR Indexer Framework. This will create a folder that keeps the database of the blockchain state and whenever you sync your node this data folder will be the same as the other nodes on the network.
+
+<blockquote class="warning">
+<strong>heads up</strong><br><br>
+
+A regular node runs with the archival option disabled by default in its `config.json`. This means that the node has a "garbage collector" enabled which removes old data from storage after five [epochs](/docs/concepts/epoch) or approx. 2.5 days. An [archival node](/docs/roles/integrator/exchange-integration#running-an-archival-node) means that the garbage collector is disabled and **all** of the data is kept on the node.
+
+You don't necessarily need to have an archival node for your indexer. In most cases you can live a happy life along with the garbage collector while you store necessary data in a separate database (either relational or nosql) much like we do with [NEAR Indexer for Wallet](https://github.com/near/near-indexer-for-wallet). However, an archival node may be necessary if you want to build an indexer which needs to find and save all transactions related to specific accounts from the beginning of the chain (from genesis).
+
+</blockquote>
+
+### Generating Configs
+
+> A typical setup is to generate configs for `localnet` whenever you pass `init`, and start the indexer whenever you pass `run` as a command line argument.
+
+- To start, add the following line at the the beginning of the `main()` function:
+
+```rs
+let args: Vec<String> = std::env::args().collect();
+```
+
+- Now create a `home_dir` variable that you will use in a few places. Add this line right after the previous one:
+
+```rs
+let home_dir = std::path::PathBuf::from(near_indexer::get_default_home());
+```
+
+- Next add a condition to check if you pass either `init` or `run` as an argument:
+
+```rs
+match args[1].as_str() {
+    "init" => {},
+    "run" => {},
+    _ => panic!("ERROR: You have to pass `init` or `run` arg."),
+}
+```
+
+- Inside the `init` code block add the `init_configs` function:
+
+```rs
+match args[1].as_str() {
+    "init" => near_indexer::init_configs(
+      &home_dir,
+      Some("localnet"),
+      None,
+      None,
+      1,
+      false,
+      None,
+      true,
+      None,
+    );,
+    "run" => {},
+    _ => panic!("ERROR: You have to pass `init` or `run` arg."),
+}
+
+```
+
+- For the `run` code block move the previous indexer start logic like so:
+
+```rs
+match args[1].as_str() {
+    "init" => near_indexer::init_configs(
```

Comment:
The signature of this function is very ugly and we want to do something about that to improve it before it get's published to the docs. I'll update this comment with more details asap. But it's definitely a blocker for now.